pkg/csi/service/node.go

```diff
@@ -528,6 +528,7 @@ func (s *service) NodeGetVolumeStats(
 	available, ok := (*(volMetrics.Available)).AsInt64()
 	if !ok {
 		log.Errorf("failed to fetch available bytes")
+		return nil, status.Error(codes.Unknown, "failed to fetch available bytes")
```

Comment:
The existing code was just ignoring the errors and moving on to get other metrics. So, why is that not right?

Reply:
I see the reasoning for ignoring error for some cases and not for other cases.

here is the original code

```
        available, ok := (*(volMetrics.Available)).AsInt64()
	if !ok {
		log.Errorf("failed to fetch available bytes")
	}
	capacity, ok := (*(volMetrics.Capacity)).AsInt64()
	if !ok {
		log.Errorf("failed to fetch capacity bytes")
		return nil, status.Error(codes.Unknown, "failed to fetch capacity bytes")
	}
	used, ok := (*(volMetrics.Used)).AsInt64()
	if !ok {
		log.Errorf("failed to fetch used bytes")
	}
	inodes, ok := (*(volMetrics.Inodes)).AsInt64()
	if !ok {
		log.Errorf("failed to fetch available inodes")
		return nil, status.Error(codes.Unknown, "failed to fetch available inodes")

	}
	inodesFree, ok := (*(volMetrics.InodesFree)).AsInt64()
	if !ok {
		log.Errorf("failed to fetch free inodes")
	}

	inodesUsed, ok := (*(volMetrics.InodesUsed)).AsInt64()
	if !ok {
		log.Errorf("failed to fetch used inodes")
	}
```

According to CSI Spec.

> If the volume is being used in BlockVolume mode then used and available MAY be omitted from usage field of NodeGetVolumeStatsResponse. Similarly, inode information MAY be omitted from NodeGetVolumeStatsResponse when unavailable.

as per 
https://github.com/container-storage-interface/spec/blob/master/csi.proto#L1278-L1298 `available` and `used` in the NodeGetVolumeStatsResponse->VolumeUsage is optional.

```
available -> OPTIONAL
total (capacity) -> REQUIRED
used -> OPTIONAL
unit -> REQUIRED
```

Let me fix this part of change.

Reply:
Fixed in latest commit.